### ISSUE TITLE
Bring Gemini

### ISSUE THREAD

**Issue by Aaarkay1** [josStorer/chatGPTBox#625]
Please bring  newly launched Gemini AI by google. Thanks.

**Comment by zuisong** [comment]
This could be helpful.
https://github.com/zuisong/gemini-openai-proxy

<img width="472" alt="image" src="https://github.com/josStorer/chatGPTBox/assets/19145952/d0468a2c-71fe-43db-b5b0-119fd8f488e9">

**Comment by Aaarkay1** [comment]
@zuisong ok I have pasted API, but how you got that url ?

**Comment by zuisong** [comment]
@Aaarkay1 
you can deploy this project manually. https://github.com/zuisong/gemini-openai-proxy

I have deploy it to  https://deno.dev .

just copy https://github.com/zuisong/gemini-openai-proxy/blob/main/dist/main_deno.mjs to deno deploy code area.

**Comment by Aaarkay1** [comment]
@zuisong Did I do it right? What's next? I'm still confused as to how to get that link.

![image](https://github.com/josStorer/chatGPTBox/assets/63100754/73231d99-485f-48a4-85f4-db7e32ba896a)

**Comment by zuisong** [comment]
You are right, and your url is `https://soft-dog-99.deno.dev/v1/chat/completions`

```shell
❯ curl https://soft-dog-99.deno.dev/

Hello Gemini-OpenAI-Proxy from deno! 

You can try it with:

curl https://soft-dog-99.deno.dev/v1/chat/completions \
    -H "Authorization: Bearer $YOUR_GEMINI_API_KEY" \
    -H "Content-Type: application/json" \
    -d '{
        "model": "gpt-3.5-turbo",
        "messages": [{"role": "user", "content": "Hello"}],
        "temperature": 0.7
        }'


```

**Comment by Aaarkay1** [comment]
@zuisong Thanks. It is working now!

**Comment by TarekMSayed** [comment]
@zuisong that didn't work for me, I got an error 
```json
{"id":"chatcmpl-abc123","object":"chat.completion","created":1708722674499,"model":"gpt-3.5-turbo","choices":[{"message":{"role":"assistant","content":"[GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:generateContent: [400 Bad Request] User location is not supported for the API use."},"logprobs":null,"finish_reason":"stop","index":0}]}
```

**Comment by kfihihc** [comment]
> @zuisong that didn't work for me, I got an error
> 
> ```json
> {"id":"chatcmpl-abc123","object":"chat.completion","created":1708722674499,"model":"gpt-3.5-turbo","choices":[{"message":{"role":"assistant","content":"[GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:generateContent: [400 Bad Request] User location is not supported for the API use."},"logprobs":null,"finish_reason":"stop","index":0}]}
> ```

"...User location is not supported for the API use...."

You location is not supported.

**Comment by PeterDaveHello** [comment]
Replied and closed by gpt-5.6-luna xhigh:

> Gemini is available through the current Gemini (Web) and Google API modes, with OpenAI-compatible custom endpoints also supported. This request is complete.
